src/test/java/test/io/ticktok/client/tick/RabbitTickerPolicyTest.java

```diff
@@ -0,0 +1,121 @@
+package test.io.ticktok.client.tick;
+
+import com.google.common.collect.ImmutableMap;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import io.ticktok.client.tick.rabbit.ChannelException;
+import io.ticktok.client.tick.TickChannel;
+import io.ticktok.client.tick.TickListener;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static io.ticktok.client.tick.rabbit.RabbitTickerPolicy.QUEUE_PARAM;
+import static io.ticktok.client.tick.rabbit.RabbitTickerPolicy.URI_PARAM;
+import static io.ticktok.client.tick.TickListener.RABBIT;
+import static java.lang.Thread.sleep;
+import static java.time.Duration.ofSeconds;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTimeoutPreemptively;
+
+class RabbitTickerPolicyTest {
+
+    private final TickListener tickListener = new TickListener();
+
+    @Test
+    void failWhenRabbitConnectionFailed() {
+        assertThrows(ChannelException.class, () -> {
+            listenOn("amqp://unknown");
+        });
+    }
+
+    private void listenOn(String uri) {
+        tickListener.forChannel(
+                TickChannel.builder()
+                        .type(RABBIT)
+                        .details(ImmutableMap.of(URI_PARAM, uri, QUEUE_PARAM, "kuku"))
+                        .build()).register(() -> {
+        });
+    }
+
+    @Test
+    void failOnInvalidUri() {
```

Comment:
shouldnt be a checked exception for this specific fail?
if not, you should assert the right msg. because now its not understandable what you are testing here

Reply:
Checked messages are bad and a thing of the past :)
Also, I don't like to test messages as I see it as configurations

Reply:
i agree about the checked messages, is an old idea that has not prove its self.
please keep in mind you choose this design. you have multiple checked exceptions.
so my remark here is just to make sure you follow this line to whole project.

Reply:
where do you see checked exception in the code? I don't understand